http_dumper.go

```diff
@@ -0,0 +1,138 @@
+package godump
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+// HTTPDebugTransport wraps a http.RoundTripper to optionally log requests and responses.
+type HTTPDebugTransport struct {
+	Transport    http.RoundTripper
+	debugEnabled bool
+	dumper       *Dumper
+}
+
+// NewHTTPDebugTransport creates a HTTPDebugTransport with debug flag cached from env.
+func NewHTTPDebugTransport(inner http.RoundTripper) *HTTPDebugTransport {
+	return &HTTPDebugTransport{
+		Transport:    inner,
+		debugEnabled: os.Getenv("HTTP_DEBUG") != "",
+		dumper:       NewDumper(),
+	}
+}
+
+// SetDebug allows toggling debug logging at runtime.
+func (t *HTTPDebugTransport) SetDebug(enabled bool) {
+	t.debugEnabled = enabled
+}
+
+// Dumper returns the Dumper instance used for logging.
+func (t *HTTPDebugTransport) Dumper() *Dumper {
+	return t.dumper
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (t *HTTPDebugTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.debugEnabled {
+		start := time.Now()
+
+		// Dump Request
+		reqDump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			reqPayload := parseHTTPDump("Request", string(reqDump))
+
+			// Perform request
+			resp, err := t.Transport.RoundTrip(req)
+			duration := time.Since(start)
+			if err != nil {
+				return resp, fmt.Errorf("HTTPDebugTransport: round trip failed: %w", err)
+			}
+
+			// Dump Response
+			respDump, err := httputil.DumpResponse(resp, true)
+			if err == nil {
+				respPayload := parseHTTPDump("Response", string(respDump))
+
+				// Combine and dump
+				transaction := map[string]any{
+					"Transaction": map[string]any{
+						"Request":  reqPayload,
+						"Response": respPayload,
+						"Duration": fmt.Sprintf("%v", duration),
+					},
+				}
+				t.dumper.Dump(transaction)
+			}
+			return resp, nil
+		}
+	}
+
+	// No debug: straight pass-through
+	resp, err := t.Transport.RoundTrip(req)
+	if err != nil {
+		return resp, fmt.Errorf("HTTPDebugTransport: pass-through round trip failed: %w", err)
+	}
+	return resp, nil
+}
```

Comment:
Usually this is defined via a middlware

It's simpler to use

Reply:
Not sure what you mean by this. Middleware is a common HTTP server concept and if this was a server dumper I'd definitely agree, but this is for client requests which typically your middleware-like intercepters are hooked at the round tripper. Please elaborate if you mean differently.